File: HDT8-SCALA VERSION/Paciente.scala
/*
* Universidad del Valle de Guatemala
* Algoritmos y Estructuras de Datos - Ing. Douglas Barrios
* Christian Fernando Morales López - 15015
* 03/10/16 - SCALA VERSION OF CLASS PACIENTE
*/

class Paciente(var nombre: String, var diagnostico: String, var prioridad: Char)
    extends Comparable[Paciente] {

  def getPrioridad(): Char = prioridad

  def getDiagnostico(): String = diagnostico

  def getNombre(): String = nombre

  override def compareTo(o: Paciente): Int = {
    val propio = this.prioridad.toInt
    val ajeno = o.getPrioridad.toInt
    if (propio >= ajeno) {
      0
    } else {
      1
    }
  }

  override def toString(): String = {
    this.nombre + "," + this.diagnostico + ", " + String.valueOf(this.prioridad)
  }
}

File: HDT8-HEAPS/src/Paciente.java
/**
 * Created by DiegoC on 02/10/2016.
 */
public class Paciente implements Comparable<Paciente> {
    public String nombre;
    public String diagnostico;
    public char prioridad;

    public Paciente(){
        
    }
    public Paciente(String nombre, String diagnostico, char prioridad){
        this.nombre = nombre;
        this.diagnostico = diagnostico;
        this.prioridad = prioridad;
    }
    public char getPrioridad() {
        return prioridad;
    }
    
    public void setter(String nombre, String diagnostico, char prioridad){
        this.nombre = nombre;
        this.diagnostico = diagnostico;
        this.prioridad = prioridad;
    }            

    public String getDiagnostico() {
        return diagnostico;
    }

    public String getNombre() {
        return nombre;
    }

    @Override
    public int compareTo(Paciente o) {
        int propio = (int)this.prioridad;
        int ajeno = (int)o.getPrioridad();
        if(propio >= ajeno){
            return 0;
        }else{
            return 1;
            }

    }
    
    public String toString(){
        return this.nombre + this.diagnostico + String.valueOf(this.prioridad);
    }
}
